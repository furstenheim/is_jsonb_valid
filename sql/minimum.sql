-- minimum validation
-- above the minimum is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '2.6');
-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '1.1');
-- below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":1.1}', '0.6');
-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":1.1}', '"x"');
-- minimum validation (explicit false exclusivity)
-- above the minimum is valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '2.6');
-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '1.1');
-- below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '0.6');
-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '"x"');
-- exclusiveMinimum validation
-- above the minimum is still valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.2');
-- boundary point is invalid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.1');
-- minimum validation with signed integer
-- negative above the minimum is valid
SELECT is_jsonb_valid('{"minimum":-2}', '-1');
-- positive above the minimum is valid
SELECT is_jsonb_valid('{"minimum":-2}', '0');
-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":-2}', '-2');
-- boundary point with float is valid
SELECT is_jsonb_valid('{"minimum":-2}', '-2');
-- float below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":-2}', '-2.0001');
-- int below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":-2}', '-3');
-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":-2}', '"x"');