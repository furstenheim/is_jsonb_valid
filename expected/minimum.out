-- minimum validation
-- above the minimum is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '2.6');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":1.1}', '1.1');
 is_jsonb_valid 
----------------
 t
(1 row)

-- below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":1.1}', '0.6');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":1.1}', '"x"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- minimum validation (explicit false exclusivity)
-- above the minimum is valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '2.6');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '1.1');
 is_jsonb_valid 
----------------
 t
(1 row)

-- below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '0.6');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":false}', '"x"');
 is_jsonb_valid 
----------------
 t
(1 row)

-- exclusiveMinimum validation
-- above the minimum is still valid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.2');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is invalid
SELECT is_jsonb_valid('{"minimum":1.1,"exclusiveMinimum":true}', '1.1');
 is_jsonb_valid 
----------------
 f
(1 row)

-- minimum validation with signed integer
-- negative above the minimum is valid
SELECT is_jsonb_valid('{"minimum":-2}', '-1');
 is_jsonb_valid 
----------------
 t
(1 row)

-- positive above the minimum is valid
SELECT is_jsonb_valid('{"minimum":-2}', '0');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point is valid
SELECT is_jsonb_valid('{"minimum":-2}', '-2');
 is_jsonb_valid 
----------------
 t
(1 row)

-- boundary point with float is valid
SELECT is_jsonb_valid('{"minimum":-2}', '-2');
 is_jsonb_valid 
----------------
 t
(1 row)

-- float below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":-2}', '-2.0001');
 is_jsonb_valid 
----------------
 f
(1 row)

-- int below the minimum is invalid
SELECT is_jsonb_valid('{"minimum":-2}', '-3');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores non-numbers
SELECT is_jsonb_valid('{"minimum":-2}', '"x"');
 is_jsonb_valid 
----------------
 t
(1 row)

